biojava-structure/src/main/java/org/biojava/nbio/structure/io/cif/AbstractCifFileSupplier.java

```diff
@@ -92,6 +90,49 @@ protected CifFile getInternal(Structure structure, List<WrappedAtom> wrappedAtom
                     .leaveCategory();
         }
 
+        if (entityInfos != null) {
+
+            String[] entityIds = new String[entityInfos.size()];
+            String[] entityTypes = new String[entityInfos.size()];
+            String[] entityDescriptions = new String[entityInfos.size()];
+
+            for (int i=0; i<entityInfos.size(); i++) {
+                EntityInfo e = entityInfos.get(i);
+                entityIds[i] = Integer.toString(e.getMolId());
+                entityTypes[i] = e.getType().getEntityType();
+                entityDescriptions[i] = e.getDescription() == null? "?" : e.getDescription();
+            }
+
+            String[] polyEntityIds = entityInfos.stream().filter(e -> e.getType() == EntityType.POLYMER).map(e -> Integer.toString(e.getMolId())).collect(Collectors.toList()).toArray(new String[]{});
```

Comment:
Streams have a `toArray` method as well. `.collect(Collectors.toList()).toArray(new String[]{})` can be simplified to `.toArray(String[]::new)`, effectively skipping the creation of a `List`.

Reply:
Nice one, thanks. Added a commit. 

And shame to IntelliJ for not suggesting this!